src/content/en/2021/capabilities.md

```diff
@@ -1,20 +1,368 @@
+} catch (err) {
+  console.log(err);
+}
+```
+
+{{ figure_markup(
+caption="Desktop websites use the Web USB API",
+content="71",
+classes="big-number",
+sheets_gid="2077755325",
+sql_file="fugu.sql"
+)
+}}
+
+The API is generally available on Chromium-based browsers on Chrome OS, Android, macOS, and Windows starting from version 56. On Linux, the API is provided behind a flag. 71 desktop and 45 mobile sites make use of this capability. For instance, the <a hreflang="en" href="https://web.brewfather.app/">Brewfather</a> PWA targeted at home-brewers allows them to send a beer recipe wirelessly over to a Bluetooth-enabled brewing system. Again, all without installing any third-party software.
+
+{{ figure_markup(
+image="web-bluetooth.png",
+caption='The Brewfather app uses Web Bluetooth to send recipes to a brew controller',
+description='Screenshot showing a web application that displays a beer recipe.',
+width=699,
+height=440
+) }}
+
+### Web Serial API
+The <a hreflang="en" href="https://web.dev/serial/">Web Serial API</a> allows you to connect with serial devices such as microcontrollers. To do so, call the `navigator.serial.requestPort()` method. You can optionally pass in a method to filter the device list. The browser shows a device picker where the user can choose a device. Next, you can open the connection by calling the port's `open()` method.
+
+```js
+try {
+  const port = await navigator.serial.requestPort();
+  await port.open({ baudRate: 9600 });
+} catch (err) {
+  console.log(err);
+}
+```
+
+{{ figure_markup(
+caption="Desktop websites use the Web Serial API",
+content="15",
+classes="big-number",
+sheets_gid="2077755325",
+sql_file="fugu.sql"
+)
+}}
+
+This capability is relatively new, as it shipped with Chromium 89 in March 2021. Currently, 15 desktop and 14 mobile sites use the Web Serial API, including the <a hreflang="en" href="https://duino.app/">Duino App</a> that allows you to develop programs for Arduino and ESP microcontrollers right in your browser. They are compiled on a remote server and then uploaded to a connected board via the Web Serial API.
+
+{{ figure_markup(
+image="web-serial.png",
+caption='The Duino app is a web-based IDE that uses Web Serial to upload programs to Arduino microcontrollers',
+description='Screenshot showing a web-based code editor application compiling code.',
+width=699,
+height=440
+) }}
+
+### Generic Sensor API
+Finally, the <a hreflang="en" href="https://web.dev/generic-sensor/">Generic Sensor API</a> allows you to read sensor data from the device's sensors, such as the accelerometer, gyroscope, or orientation sensor. To access a sensor, you create a new instance of a sensor class, e.g., `Accelerometer`. The constructor takes a configuration object with the requested frequency. By attaching to the `onreading` and `onerror` events, you can get notified for updated sensor values, or errors respectively. Finally, you need to start the reading by calling the `start()` method.
```

Comment:
It's about the device's sensors (i.e., accelerometer, or gyroscope), I hoped the first sentence would make that clear. Would it help to rephrase it like this maybe?

```suggestion
While the previous APIs provide access to data transmission interfaces, the <a hreflang="en" href="https://web.dev/generic-sensor/">Generic Sensor API</a> allows you to read sensor data from the device's sensors, such as the accelerometer, gyroscope, or orientation sensor. To access a sensor, you create a new instance of a sensor class, e.g., `Accelerometer`. The constructor takes a configuration object with the requested frequency. By attaching to the `onreading` and `onerror` events, you can get notified for updated sensor values, or errors respectively. Finally, you need to start the reading by calling the `start()` method.
```

Reply:
Ah I got you. So it's like devices and sensors, rather than direct serial port access to get to those devices.

Hmmmm.... can't think of a better way of phrasing it to be honest. Think I prefer your original wording. Will just leave it as is and resolve this.